### DISCUSSION TITLE
Broken DAG while calling Slack providers module

### DISCUSSION THREAD

**Discussion by thomaspanji** [apache/airflow#21910] [Q&A] [answered]
Hi, I am using Airflow version 2.2.3 from Google Composer. I'd like to use `SlackAPIPostOperator` by calling `from airflow.providers.slack.operators.slack import SlackAPIPostOperator` but Airflow sends error while loading the module:

> Broken DAG: [/home/airflow/gcs/dags/test_dag.py] Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/airflow/gcs/dags/test_dag.py", line 14, in <module>
    from airflow.providers.slack.operators.slack import SlackAPIPostOperator
ModuleNotFoundError: No module named 'airflow.providers.slack'

Is anything wrong with my code? Or is Slack operator not supported in Airflow 2.2.3?

**Comment by josh-fell** [comment, marked as answer]
Do you have the [Slack provider](https://pypi.org/project/apache-airflow-providers-slack/) installed in your environment? Providers are decoupled from core Airflow and need to be installed separately into Airflow environments.

It doesn't look like the Slack provider is installed out-of-the-box with Google Cloud Composer's Airflow 2.2.3 versions: https://cloud.google.com/composer/docs/concepts/versioning/composer-versions

**Reply by thomaspanji** [reply]
Ah you're right! I'm new to Google Composer. Thanks for the answer.
